src/key_io.cpp

```diff
@@ -270,36 +270,58 @@ std::string EncodePaymentAddress(const libzcash::PaymentAddress& zaddr)
     return boost::apply_visitor(PaymentAddressEncoder(Params()), zaddr);
 }
 
-libzcash::PaymentAddress DecodePaymentAddress(const std::string& str)
+template<typename T1, typename T2, typename T3 = T2>
+T1 DecodeAny(
+    CChainParams::Base58Type type,
+    size_t size,
+    const std::string& str,
```

Comment:
Make this the first argument to `DecodeAny` (so the pairs are all next to each other in calls to `DecodeAny`).